### PR TITLE
fix(ComposedModal): fix focus trap and refocus to trigger element when closed

### PR DESCRIPTION
**Pull request by annawen1**
Closes  
- #11046 
- #11059

Gets changes from https://github.com/carbon-design-system/carbon/pull/11350 and https://github.com/carbon-design-system/carbon/pull/11390 into the `v10` branch.

https://user-images.githubusercontent.com/54281166/168654043-4186899a-04b1-4549-a181-6ac60041da5f.mov

#### Changelog

**Changed**

- wrap children in another div to get focus wrapping to work
- set `aria-hidden` when modal is closed to prevent VO from reading out modal content when closed
- pass logic to re-focus to trigger element `onClose` in the ComposedModal story

#### Testing / Reviewing

test using VO on iOS
- go to ComposedModal, and swipe through modal content, making sure focus is trapped within modal
- also to go `State Manager` story to ensure when modal is closed, focus goes back to the "Launch composed modal" button

### CODE REVIEW

File: packages/styles/scss/components/modal/_modal.scss
Comment (annawen1) [inline packages/styles/scss/components/modal/_modal.scss:112]: @tay1orjones should the new styles be added to both `packages/styles/scss/components/modal/_modal.scss` and `packages/components/src/components/modal/_modal.scss`?

File: packages/components/src/components/modal/_modal.scss
Comment (joshblack) [inline packages/components/src/components/modal/_modal.scss:111]: Just wanted to double-check, do we need to offer any fallbacks for IE11 support or should this work as expected?

**Review by joshblack — APPROVED** [review]

File: packages/styles/scss/components/modal/_modal.scss
Comment (joshblack) [reply packages/styles/scss/components/modal/_modal.scss:112]: You may be able to get away with only `packages/components` in v10 at this point since the styles package is private, curious what you think @tay1orjones 👀  Definitely doesn't hurt having it in both places though (especially if that makes porting things back easier)

File: packages/components/src/components/modal/_modal.scss
Comment (annawen1) [reply packages/components/src/components/modal/_modal.scss:111]: @joshblack oh I didn't think about that, are you guys still supporting IE11? the dotcom team has dropped IE from our supported browser list.

Comment (joshblack) [reply packages/components/src/components/modal/_modal.scss:111]: @annawen1 it shouldn't be a problem in v11 but technically we do still support it for v10. If the file is already using non-IE11 supported features though it's definitely okay to ship as-is which is why I was curious if that was already in place

File: packages/styles/scss/components/modal/_modal.scss
Comment (tay1orjones) [reply packages/styles/scss/components/modal/_modal.scss:112]: Yeah, same thought. Having the same in both spots is preferred/nice but not necessary.

**Review by tay1orjones — APPROVED** [review]
Looks good pending the comments from @joshblack 👍 

Once merged, does this need to go out in a v10 patch now, or can it wait until our release next week?

**Comment by annawen1** [comment]
> Looks good pending the comments from @joshblack 👍
> 
> Once merged, does this need to go out in a v10 patch now, or can it wait until our release next week?

@tay1orjones I think we can sit on it till the release next week!

**Pull request merged**
